golang/pkg/dagent/utils/docker.go

```diff
@@ -294,22 +294,19 @@ func DeployImage(ctx context.Context,
 
 	mountList := buildMountList(cfg, dog, deployImageRequest)
 
-	matchedContainer, err := dockerHelper.GetContainerByName(ctx, dog, containerName)
+	matchedContainer, err := dockerHelper.GetContainerByName(ctx, dog, containerName, true)
```

Comment:
It's not an error if there's no container before deploying the new one. We only need this to report the state of the old container. This is especially important when deploying the very first version of something since the containers won't exist at that point.

I think you could set `errorReport` to false and check `matchedContainer` before writing the state using dogger.

Reply:
fixed